builder/azure/arm/step_create_resource_group.go

```diff
@@ -15,7 +15,7 @@ type StepCreateResourceGroup struct {
 	create func(resourceGroupName string, location string, tags *map[string]*string) error
 	say    func(message string)
 	error  func(e error)
-	exists func(resourceGroupName string) (bool, error)
```

Comment:
The better place to make this update is in builder.go.  The code already uses the GroupsClient, so you should have everything you need (including an example) to update the location.  I would rather make the change there instead of in the step, so the logic can be kept smaller.

I think this will make the overall change smaller too.

Reply:
That does make it smaller. I was thinking we want to keep it to just one existence check but i can see the advantage in checking earlier as well. 

It's a strange relationship between Config and the stateBag for a newbie and where and why different things are validated in different locations.